syntax = "proto3";

service PostService {
    rpc CreatePost (CreatePostRequest) returns (CreatePostResponse);
    rpc UpdatePost (UpdatePostRequest) returns (UpdatePostResponse);
    rpc DeletePost (DeletePostRequest) returns (DeletePostResponse);
    rpc GetPost (GetPostRequest) returns (GetPostResponse);
    rpc ListPosts (ListPostsRequest) returns (ListPostsResponse);
}

message Post {
    int64 id = 1;
    int64 user_id = 2;
    string title = 3;
    string content = 4;
    int64 timestamp = 5;
}

message CreatePostRequest {
    int64 user_id = 1;
    string title = 2;
    string content = 3;
}

message CreatePostResponse {
    Post post = 1;
}

message UpdatePostRequest {
    int64 id = 1;
    int64 user_id = 2;
    string title = 3;
    string content = 4;
}

message UpdatePostResponse {
    Post post = 1;
}

message DeletePostRequest {
    int64 id = 1;
    int64 user_id = 2;
}

message DeletePostResponse {
    bool success = 1;
}

message GetPostRequest {
    int64 id = 1;
}

message GetPostResponse {
    Post post = 1;
}

message ListPostsRequest {
    int64 user_id = 1;
}

message ListPostsResponse {
    repeated Post posts = 1;
}
